pages/instructorGradebook/instructorGradebook.sql

```diff
@@ -49,7 +61,10 @@ scores AS (
         course_users AS u
         CROSS JOIN course_assessments AS a
         LEFT JOIN group_users AS gu ON (u.user_id = gu.user_id)
-        LEFT JOIN course_scores AS s ON (((s.user_id = u.user_id)OR(s.group_id = gu.group_id)) AND s.assessment_id = a.id)
+        LEFT JOIN groups AS gr ON (gr.id = gu.group_id)
+        LEFT JOIN course_scores AS s ON (((s.user_id = u.user_id) OR (s.group_id = gu.group_id)) AND s.assessment_id = a.id)
+     WHERE 
+         gr.deleted_at IS NULL
```

Comment:
Just to confirm, this part is the main fix for the bug, right? And the rest of the changes are purely for reformatting?

Reply:
Yeah. That bug occurs when old groups deleted and the student join a new group, I did not check old groups' delete_at.

Reply:
While this is a bug, I don't think it's actually the bug here. The problems with the gradebook are showing up for courses that aren't even using groups at all. I think the problem is that you are JOINing over all `group_users`, even if they are unrelated to the current course_instance or assessment.

Reply:
I submitted PR #2965 which I think is a fix for this bug. But please check that PR!